prompts/few_shot.go

```diff
@@ -0,0 +1,189 @@
+package prompts
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+var (
+	// ErrNoExample is returned when none of the Examples and ExampleSelector are provided.
+	ErrNoExample = errors.New("no example is provided")
+	// ErrExamplesAndExampleSelectorProvided is returned when there are no Examples and ExampleSelector.
+	ErrExamplesAndExampleSelectorProvided = errors.New("only one of 'Examples' and 'example_selector' should be" +
+		" provided")
+)
+
+// FewShotPrompt contains fields for a few-shot prompt.
+type FewShotPrompt struct {
+	// Examples to format into the prompt. Either this or ExamplePrompt should be provided.
+	Examples []map[string]string
+	// ExampleSelector to choose the examples to format into the prompt. Either this or Examples should be provided.
+	ExampleSelector ExampleSelector
+	// ExamplePrompt is used to format an individual example.
+	ExamplePrompt PromptTemplate
+	// A prompt template string to put before the examples.
+	Prefix string
+	// A prompt template string to put after the examples.
+	Suffix string
+	// A list of the names of the variables the prompt template expects.
+	InputVariables map[string]any
+	// Represents a map of variable names to values or functions that return values. If the value is a function, it will
+	// be called when the prompt template is rendered.
+	PartialVariables map[string]any
+	// String separator used to join the prefix, the examples, and suffix.
+	ExampleSeparator string
+	// The format of the prompt template. Options are: 'f-string', 'jinja2'.
+	TemplateFormat TemplateFormat
+	// Whether to try validating the template.
+	ValidateTemplate bool
+}
+
+// FewShotCallOptions contains optional fields for FewShotPrompt.
+type FewShotCallOptions struct {
```

Comment:
if we're putting multiple field sin here then the options argument to New shouldn't be variadic

Reply:
I wanted to let developers not provide any option instead of passing an empty CallOption instance. I guess we have the same with the [llms.options](https://github.com/tmc/langchaingo/blob/main/llms/options.go).

Reply:
I don't think having options to the constructor is necessary for this.

Reply:
So you mean removing both the variadic argument and FewShotCallOptions?

Reply:
What about default values?


```
		ExampleSeparator: "\n\n",
		TemplateFormat:   TemplateFormatGoTemplate,
		ValidateTemplate: true,
```
these are their default values. I can set their type as a pointer to let developers set nil and I set their default. I want them to be able to use default values without explicitly setting them. Thoughts?